Backport from PHP 5.5.0 and 5.5.19

Adapted from:

From e69f987948982d4259a574ca824398c26153bf42 Mon Sep 17 00:00:00 2001
From: Pierrick Charron <pierrick@php.net>
Date: Thu, 1 Dec 2011 21:48:07 +0000
Subject: [PATCH] Clean / Improve the curl extension # NEWS file will come soon

From 2b5bffe6c70bc00ebe57390f48ef7569e401d2d3 Mon Sep 17 00:00:00 2001
From: Rasmus Lerdorf <rasmus@php.net>
Date: Thu, 16 Oct 2014 21:25:29 -0700
Subject: [PATCH] TLS 1.0, 1.1 and 1.2 Curl constants - bug #68247

Macro available in upstream curl > 7.34
Macro available since curl-7.19.7-43.el6 see https://bugzilla.redhat.com/1012136


diff -up a/ext/curl/interface.c.old b/ext/curl/interface.c
--- a/ext/curl/interface.c.old	2015-12-14 11:29:34.591570003 +0100
+++ b/ext/curl/interface.c	2015-12-14 11:38:42.366016986 +0100
@@ -595,6 +595,16 @@ PHP_MINIT_FUNCTION(curl)
 	REGISTER_CURL_CONSTANT(CURLOPT_SSL_VERIFYHOST);
 	REGISTER_CURL_CONSTANT(CURLOPT_COOKIEFILE);
 	REGISTER_CURL_CONSTANT(CURLOPT_SSLVERSION);
+
+        /* Curl SSL Version constants (CURLOPT_SSLVERSION) */
+	REGISTER_CURL_CONSTANT(CURL_SSLVERSION_DEFAULT);
+	REGISTER_CURL_CONSTANT(CURL_SSLVERSION_SSLv2);
+	REGISTER_CURL_CONSTANT(CURL_SSLVERSION_SSLv3);
+	REGISTER_CURL_CONSTANT(CURL_SSLVERSION_TLSv1);
+	REGISTER_CURL_CONSTANT(CURL_SSLVERSION_TLSv1_0);
+	REGISTER_CURL_CONSTANT(CURL_SSLVERSION_TLSv1_1);
+	REGISTER_CURL_CONSTANT(CURL_SSLVERSION_TLSv1_2);
+
 	REGISTER_CURL_CONSTANT(CURLOPT_TIMECONDITION);
 	REGISTER_CURL_CONSTANT(CURLOPT_TIMEVALUE);
 	REGISTER_CURL_CONSTANT(CURLOPT_CUSTOMREQUEST);
